storage/database/db_manager.go

```diff
@@ -934,12 +939,31 @@ func (dbm *databaseManager) ReadHeadBlockHash() common.Hash {
 	return common.BytesToHash(data)
 }
 
+// Block Backup Hash operations.
+func (dbm *databaseManager) ReadHeadBlockBackupHash() common.Hash {
+	db := dbm.getDatabase(headerDB)
+	data, _ := db.Get(headBlockBackupKey)
+	if len(data) == 0 {
+		return common.Hash{}
+	}
+	return common.BytesToHash(data)
+}
+
 // WriteHeadBlockHash stores the head block's hash.
 func (dbm *databaseManager) WriteHeadBlockHash(hash common.Hash) {
+	backupHashes[idx%128] = hash
```

Comment:
It would be better to add a constant instead of a magic number 128.

Reply:
Changes to constant value